hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/RPC.java

```diff
@@ -419,7 +419,7 @@ public RpcErrorCodeProto getRpcErrorCodeProto() {
         ioe = nrthe;
       }
       // check if timed out
-      if (Time.now()-timeout >= startTime) {
+      if (Time.monotonicNow() - timeout >= startTime) {
```

Comment:
Similarly, I think `Time.monotonicNow() - startTime >= timeout` would be better here.